fdbserver/workloads/DataLossRecovery.actor.cpp

```diff
@@ -0,0 +1,328 @@
+	  : TestWorkload(wcx), startMoveKeysParallelismLock(1), finishMoveKeysParallelismLock(1), enabled(!clientId),
+	    pass(true) {}
+
+	void validationFailed(KeyRef key, Optional<Value>& expectedValue, Optional<Value>& actualValue) {
+		TraceEvent(SevError, "TestFailed")
+		    .detail("Key", key.toString())
+		    .detail("ExpectedValue", printValue(expectedValue))
+		    .detail("ActualValue", printValue(actualValue));
+		pass = false;
+	}
+
+	std::string description() const override { return "DataLossRecovery"; }
+
+	Future<Void> setup(Database const& cx) override { return Void(); }
+
+	Future<Void> start(Database const& cx) override {
+		if (!enabled) {
+			return Void();
+		}
+		return _start(this, cx);
+	}
+
+	ACTOR Future<Void> _start(DataLossRecoveryWorkload* self, Database cx) {
+		state Key key = "TestKey"_sr;
+		state Key endKey = "TestKey0"_sr;
+		state Value oldValue = "TestValue"_sr;
+		state Value newValue = "TestNewValue"_sr;
+
+		wait(self->writeAndVerify(self, cx, key, oldValue));
+
+		state Optional<StorageServerInterface> ssi = wait(self->getRandomStorageServer(cx));
+		ASSERT(ssi.present());
+
+		// Move [key, endKey) to team: {address}.
+		state std::vector<NetworkAddress> addresses;
+		addresses.push_back(ssi.get().address());
+		std::cout << "Moving." << std::endl;
+		wait(moveShard(cx->getConnectionFile(), KeyRangeRef(key, endKey), addresses));
+
+		Transaction tr(cx);
+		Standalone<VectorRef<const char*>> adds = wait(tr.getAddressesForKey(key));
+		ASSERT(adds.size() == 1);
+		ASSERT(adds[0] == ssi.get().address().toString());
+		std::cout << "Moved to: " << adds[0] << "(" << ssi.get().uniqueID.toString() << ")" << std::endl;
+
+		wait(self->readAndVerify(self, cx, key, oldValue));
+
+		// Kill team {address}, and expect read to timeout.
+		self->killProcess(self, ssi.get().address());
+		wait(self->readAndVerify(self, cx, key, "Timeout"_sr));
+
+		// Reenable DD and exclude address as fail, so that [key, endKey) will be dropped and moved to a new team.
+		// Expect read to return 'value not found'.
+		int ignore = wait(setDDMode(cx, 1));
+		wait(self->exclude(self, cx, key, ssi.get().address()));
+		wait(self->readAndVerify(self, cx, key, Optional<Value>()));
+
+		// Write will scceed.
+		wait(self->writeAndVerify(self, cx, key, newValue));
+
+		// **********************************************************************
+
+		state Optional<StorageServerInterface> ssi2 = wait(self->getRandomStorageServer(cx));
+		ASSERT(ssi2.present());
+		ASSERT(ssi2 != ssi);
+
+		// Move [\xff, \xff\xff) to team: {ssi}.
+		addresses.clear();
+		addresses.push_back(ssi2.get().address());
+		std::cout << "Moving." << std::endl;
+		wait(moveShard(cx->getConnectionFile(), systemKeys, addresses));
+
+		// Kill team {address}, and expect read to timeout.
+		self->killProcess(self, ssi2.get().address());
+		std::cout << "Killed process: " << ssi2.get().address().toString() << "(" << ssi2.get().toString() << ")"
```

Comment:
How did it kill all 3 replicas by only killing one process?
Did I miss something?